core/src/main/scala/org/apache/spark/util/SparkUncaughtExceptionHandler.scala

```diff
@@ -46,6 +46,18 @@ private[spark] object SparkUncaughtExceptionHandler
     }
   }
 
+  def log(thread: Thread, exception: Throwable): Unit = {
+    val errMsg = "Uncaught exception in thread "
+    val inShutdownMsg = "[Container in shutdown] "
```

Comment:
minor: less code if done like this:

    val inShutdownMessage = if (ShutdownHookManager.inShutdown()) "[Container in shutdown] " else ""
    logError(inShutdownMsg + errMsg + thread, exception)